qiskit_research/utils/dynamical_decoupling.py

```diff
@@ -182,3 +192,51 @@ def add_pulse_calibrations(
                 pulse.play(inverted_pulse, x_instruction.channel)
             for circ in circuits:
                 circ.add_calibration("ym", [qubit], sched)
+
+
+def get_urdd_phis(urdd_pulse_num: int = 4) -> Sequence[float]:
+    """Gets \\phi_k values for n pulse UR sequence"""
+    if urdd_pulse_num % 2 == 1:
+        raise ValueError("urdd_pulse_num must be even")
+    if urdd_pulse_num < 4:
+        raise ValueError("urdd_pulse_num must be >= 4")
+
+    # get capital Phi value
+    if urdd_pulse_num % 4 == 0:
+        m_divisor = int(urdd_pulse_num / 4)
+        big_phi = np.pi / m_divisor
+    else:
+        m_divisor = int((urdd_pulse_num - 2) / 4)
+        big_phi = (2 * m_divisor * np.pi) / (2 * m_divisor + 1)
+
+    # keep track of unique phi added; we choose phi2 = big_phi by convention--
+    # only real requirement is (n * big_phi = 2pi * j for j int)
+    unique_phi = [0, big_phi]
+    # map each phi in [phis] to location (by index) of corresponding [unique_phi]
+    phi_indices = [0, 1]
+    # populate remaining phi values
+    for kk in range(3, urdd_pulse_num + 1):
+        phi_k = (kk * (kk - 1) * big_phi) / 2
+        # values only matter modulo 2 pi
+        phi_k = (phi_k) % (2 * np.pi)
+        if np.isclose(phi_k, 0):
+            phi_k = 0
+        elif np.isclose(phi_k, 2 * np.pi):
+            phi_k = 0
+
+        added_new = False
+        for idx, u_phi in enumerate(unique_phi):
+            if np.isclose(u_phi, phi_k, atol=0.001):
```

Comment:
Instead of setting seemingly arbitrary values for `atol` and the like, would it make sense to pass them as parameters to the function?

Reply:
That's a good question, as @bibekpokharel wrote this code (and deserves credit for his contribution), perhaps he can answer this.

Reply:
It should be fine to pass this as a parameter, but that might be unnecessary. This line asserts that u_phi and phi_k are equal upto the precision defined in the single qubit rotation in qiskit. If this value is defined elsewhere in qiskit, it can be inherited from there.

Reply:
I spoke with @bibekpokharel offline, and we can probably use